File: src/main/scala/Eulers_Problems/Java/MultipleOf3And5.java
/**
 * @author Ram Saran Vuppuluri
 * Problem statement
 * @see https://www.hackerrank.com/contests/projecteuler/challenges/euler001/problem
 */
package Eulers_Problems.Java;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class MultipleOf3And5 {
    public static void main(String... args) {
        Scanner sin = new Scanner(System.in);
        int numberOfParameters = sin.nextInt();
        List<Long> sumList = new ArrayList<Long>();
        for (int i = 0; i < numberOfParameters; i++) {
            long inputLong = sin.nextLong() - 1;
            sumList.add(sumCalc(inputLong));
        }

        for (long longVal : sumList) {
            System.out.println(longVal);
        }
    }

    private static long sumCalc(long inputLong) {
        return (sumOfFirstN(inputLong / 3) * 3) + (sumOfFirstN(inputLong / 5) * 5) - (sumOfFirstN(inputLong / 15) * 15);
    }

    /**
     * Using Mathematical Formula n*(n+1)/2
     *
     * @param n
     * @return
     */
    private static long sumOfFirstN(long n) {
        return n * (n + 1) / 2;
    }
}

File: src/main/scala/Eulers_Problems/Scala/MultipleOf3And5.scala
/**
  * @author Ram Saran Vuppuluri
  *         Problem statement
  * @see https://www.hackerrank.com/contests/projecteuler/challenges/euler001/problem
  */
package Eulers_Problems.Scala

import java.util
import java.util.Scanner

object MultipleOf3And5 extends App {
  val scanner = new Scanner(System.in)
  val numberOfIteration = scanner.nextLong()
  val listOfNumbers: util.ArrayList[Long] = new util.ArrayList[Long]()
  for (i <- BigInt(1) to numberOfIteration) {
    listOfNumbers.add(scanner.nextLong() - 1)
  }

  listOfNumbers.forEach(printSum(_))

  def printSum(longNumber: Long): Unit = {
    val longOne: Long = 1
    println((((longOne to longNumber / 3).sum) * 3) + (((longOne to longNumber / 5).sum) * 5) - (((longOne to longNumber / 15).sum)
      * 15))
  }
}